src/commands.rs

```diff
@@ -458,43 +458,57 @@ fn handle_compile_response<T>(mut creator: T,
         CompileResponse::CompileStarted(_) => {
             debug!("Server sent CompileStarted");
             // Wait for CompileFinished.
-            let mut res = conn.read_one_response().chain_err(|| {
-                //TODO: something better here?
-                "error reading compile response from server"
-            })?;
-            if res.has_compile_finished() {
-                trace!("Server sent CompileFinished");
-                handle_compile_finished(res.take_compile_finished(),
-                                        stdout, stderr)
-            } else {
-                bail!("unexpected response from server")
+            match conn.read_one_response() {
+                Ok(mut res) => {
+                    if res.has_compile_finished() {
+                        trace!("Server sent CompileFinished");
+                        return handle_compile_finished(res.take_compile_finished(),
+                                                       stdout, stderr)
+                    } else {
+                        bail!("unexpected response from server")
+                    }
+                }
+
+                // Currently the shutdown behavior of the remote sccache server
+                // is to wait at most N seconds for all active connections to
+                // finish and then close everything. If we get unlucky and don't
+                // get a response then we just forge ahead locally and run the
+                // compilation ourselves.
+                Err(ProtobufError::IoError(ref e))
+                    if e.kind() == io::ErrorKind::UnexpectedEof => {}
```

Comment:
Can you put a log message here so it's easy to figure out when this happens? Actually I think even straight printing something to stderr would be OK in this case, it feels like a bug if we hit this in practice.